src/WebJobs.Script/Host/WorkerFunctionMetadataProvider.cs

```diff
@@ -79,7 +79,15 @@ public async Task<FunctionMetadataResult> GetFunctionMetadataAsync(IEnumerable<R
                     await _channelManager.InitializeChannelAsync(workerConfigs, _workerRuntime);
                     channels = _channelManager.GetChannels(_workerRuntime);
                 }
-                // start up GRPC channels
+
+                if (channels is null)
+                {
+                    _logger.LogWarning("No initialized language worker channel found for runtime: {workerRuntime}. "
+                        + "There is likely an issue with the worker not being able to start.", _workerRuntime);
+                    // Q: Instead of the null reference exception, we now fallback to default (host) metadata indexing.
+                    // Is this the right thing to do? Should we throw an exception instead?
+                    return new FunctionMetadataResult(useDefaultMetadataIndexing: true, _functions);
```

Comment:
@brettsam @soninaren @kshyju if there's something wrong with the worker, this won't do anything - regardless of indexing, the worker is corrupted. Honestly, I'm not convinced we should even be making any changes to this, or at best we can throw our own exception that isn't a "null ref"

Reply:
Discussed offline: throwing an exception as we are expecting to fail here